src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractWalletFeature.cs

```diff
@@ -1,139 +1,66 @@
-
-            nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component, this.GetType().Name);
-            nodeStats.RegisterStats(this.AddInlineStats, StatsType.Inline, this.GetType().Name);
-        }
-
-        private void AddComponentStats(StringBuilder log)
-        {
-            IEnumerable<string> walletNamesSQL = this.walletManager.GetWalletsNames();
-
-            if (walletNamesSQL.Any())
-            {
-                log.AppendLine(">> Wallets");
-
-                var walletManager = (WalletManager)this.walletManager;
-
-                foreach (string walletName in walletNamesSQL)
-                {
-                    foreach (AccountBalance accountBalance in walletManager.GetBalances(walletName))
-                    {
-                        log.AppendLine($"{walletName}/{accountBalance.Account.Name}".PadRight(LoggingConfiguration.ColumnLength) + $": Confirmed balance: {accountBalance.AmountConfirmed}".PadRight(LoggingConfiguration.ColumnLength + 20) + $" Unconfirmed balance: {accountBalance.AmountUnconfirmed}");
-                    }
-                }
-            }
-        }
-
-        private void AddInlineStats(StringBuilder log)
-        {
-            if (this.walletManager is WalletManager walletManager)
-            {
-                int height = walletManager.LastBlockHeight();
-                ChainedHeader block = this.chainIndexer.GetHeader(height);
-                uint256 hashBlock = block == null ? 0 : block.HashBlock;
-
-                if (this.walletManager.ContainsWallets)
-                    log.AppendLine("Wallet Height".PadRight(LoggingConfiguration.ColumnLength) + $": {height}".PadRight(10) + $"(Hash: {hashBlock})");
-                else
-                    log.AppendLine("Wallet Height".PadRight(LoggingConfiguration.ColumnLength) + ": No Wallet");
-
-                log.AppendLine("");
-            }
         }
 
         /// <inheritdoc />
         public override Task InitializeAsync()
         {
-            this.walletManager.Start();
-            this.walletSyncManager.Start();
-
-            this.connectionManager.Parameters.TemplateBehaviors.Add(this.broadcasterBehavior);
-
             this.logger.LogInformation("Smart Contract Feature Wallet Injected.");
             return Task.CompletedTask;
         }
 
         /// <inheritdoc />
         public override void Dispose()
         {
-            this.walletManager.Stop();
-            this.walletSyncManager.Stop();
         }
     }
 
     public static partial class IFullNodeBuilderExtensions
     {
-        public static IFullNodeBuilder UseSmartContractWallet(this IFullNodeBuilder fullNodeBuilder)
+        public static IFullNodeBuilder UseSmartContractWallet(this IFullNodeBuilder fullNodeBuilder, bool addVanillaWallet = true)
         {
             LoggingConfiguration.RegisterFeatureNamespace<WalletFeature>("smart contract wallet");
 
+            if (addVanillaWallet)
```

Comment:
Wont this then always be true? Why would we call this with false?

Reply:
```
.UseColdStakingWallet()
.UseSmartContractWallet(false)
```